x-pack/plugins/integration_assistant/server/routes/routes_util.ts

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { GraphRecursionError } from '@langchain/langgraph';
+import { RecursionLimitError } from '../lib/errors';
+
+/**
+ * Handles errors that occur during the execution of a function.
+ * If the error is an instance of GraphRecursionError, it throws a RecursionLimitError with the same message and error code.
+ * Otherwise, it rethrows the original error.
+ *
+ * @param err - The error that occurred.
+ * @param errorCode - The error code associated with the error.
+ * @throws {RecursionLimitError} If the error is an instance of GraphRecursionError.
+ * @throws {Error} The original error.
+ */
+export function handleCustomErrors(err: Error, errorCode: string) {
```

Comment:
NIT: the `errorCode` here only applies when there's a recursion error, we could name it `recursionErrorCode` and define it with the proper type:
```suggestion
export function handleCustomErrors(err: Error, recursionErrorCode:  ErrorCode.RECURSION_LIMIT | ErrorCode.RECURSION_LIMIT_ANALYZE_LOGS) {
```